# Examining App Development Potential Based on Data for Apps Available on both Apple and Android

In looking for datasets to dive into, I thought that these two from kaggle proved pretty interesting: https://www.kaggle.com/lava18/google-play-store-apps and https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps. 

These datasets had information from both google play and the app store regarding categories, ratings, genres, and pricing for all of the apps on each store.  If an app-making company was so inclined, it could use this information to gain market insight into the basis for a successful app.  

For this project, I decided to look at that data as if I was working for a company that builds free apps, with a main source of revenue being in-app ads.  This means that revenue would be gained from users engaging with those ads, which means that the more users using that app, the better.  How can I analyze this data to help gain insight into what type of free app would have the highest potential revenue source based on the features given?  Or - will the data reveal that a paid app might be more lucrative? Let's extract some info from this data and see what trends can be discovered! 

### Toggle the Raw Code On/Off If You'd Like

In [154]:
from IPython.display import HTML

HTML('''<script>
code_show=true; 
function code_toggle() {
 if (code_show){
 $('div.input').hide();
 } else {
 $('div.input').show();
 }
 code_show = !code_show
} 
$( document ).ready(code_toggle);
</script>
<form action="javascript:code_toggle()"><input type="submit" value="Click here to toggle on/off the raw code."></form>''')

# Exploring the Initial Data

Let's see what we have at first glance. 

In [111]:
import pandas as pd
import numpy as np
from pandas import DataFrame

import warnings
warnings.filterwarnings('ignore')

In [112]:
#Two files:

andr = pd.read_csv('/Users/atrain31/Desktop/Data/dq1/googleplaystore.csv')
apple = pd.read_csv('/Users/atrain31/Desktop/Data/dq1/AppleStore.csv')

Google Data:

In [113]:
andr.head()

App        Category  Rating  \
0     Photo Editor & Candy Camera & Grid & ScrapBook  ART_AND_DESIGN     4.1   
1                                Coloring book moana  ART_AND_DESIGN     3.9   
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ART_AND_DESIGN     4.7   
3                              Sketch - Draw & Paint  ART_AND_DESIGN     4.5   
4              Pixel Draw - Number Art Coloring Book  ART_AND_DESIGN     4.3   

  Reviews  Size     Installs  Type Price Content Rating  \
0     159   19M      10,000+  Free     0       Everyone   
1     967   14M     500,000+  Free     0       Everyone   
2   87510  8.7M   5,000,000+  Free     0       Everyone   
3  215644   25M  50,000,000+  Free     0           Teen   
4     967  2.8M     100,000+  Free     0       Everyone   

                      Genres      Last Updated         Current Ver  \
0               Art & Design   January 7, 2018               1.0.0   
1  Art & Design;Pretend Play  January 15, 2018               2.0.0   
2               Art & Design    August 1, 2018               1.2.4   
3               Art & Design      June 8, 2018  Varies with device   
4    Art & Design;Creativity     June 20, 2018                 1.1   

    Android Ver  
0  4.0.3 and up  
1  4.0.3 and up  
2  4.0.3 and up  
3    4.2 and up  
4    4.4 and up

Apple Data:

In [114]:
apple.head()

id               track_name  size_bytes currency  price  \
0  284882215                 Facebook   389879808      USD    0.0   
1  389801252                Instagram   113954816      USD    0.0   
2  529479190           Clash of Clans   116476928      USD    0.0   
3  420009108               Temple Run    65921024      USD    0.0   
4  284035177  Pandora - Music & Radio   130242560      USD    0.0   

   rating_count_tot  rating_count_ver  user_rating  user_rating_ver      ver  \
0           2974676               212          3.5              3.5     95.0   
1           2161558              1289          4.5              4.0    10.23   
2           2130805               579          4.5              4.5  9.24.12   
3           1724546              3842          4.5              4.0    1.6.2   
4           1126879              3594          4.0              4.5    8.4.1   

  cont_rating        prime_genre  sup_devices.num  ipadSc_urls.num  lang.num  \
0          4+  Social Networking               37                1        29   
1         12+      Photo & Video               37                0        29   
2          9+              Games               38                5        18   
3          9+              Games               40                5         1   
4         12+              Music               37                4         1   

   vpp_lic  
0        1  
1        1  
2        1  
3        1  
4        1

Google and Apple data type:

In [115]:
andr.info(), apple.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10841 entries, 0 to 10840
Data columns (total 13 columns):
App               10841 non-null object
Category          10841 non-null object
Rating            9367 non-null float64
Reviews           10841 non-null object
Size              10841 non-null object
Installs          10841 non-null object
Type              10840 non-null object
Price             10841 non-null object
Content Rating    10840 non-null object
Genres            10841 non-null object
Last Updated      10841 non-null object
Current Ver       10833 non-null object
Android Ver       10838 non-null object
dtypes: float64(1), object(12)
memory usage: 1.1+ MB
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7197 entries, 0 to 7196
Data columns (total 16 columns):
id                  7197 non-null int64
track_name          7197 non-null object
size_bytes          7197 non-null int64
currency            7197 non-null object
price               7197 non-null float64
rating_

(None, None)

So there are some interesting possiblities in this data.  Listed here are the column names (Google play, followed by the Apple app store) for each dataframe that I feel can be correlated and grouped together for analysis, followed by why they may be useful: 

App/track_name - basic identifiers for each column (the apple 'id' doesn't seem to mean anything here)

Category/prime_genre - category of each app ('Genres' seems like it would be too specific and therefore give a cloudier picture of the genres).  This would most likely be the main determinant of what type of app to create - which categories are the most popular?  Which have future potential? 

Rating/user_rating - do higher ratings equal more downloads or possibly higher price? 

Reviews/rating_count_tot  (or Installs??)- see above

(Type & Price)/price - how does pricing affect downloads and ratings?

Content Rating/cont_rating

# Cleaning the Data & Initial Analysis

Before we start, some of the kaggle comments mentioned a line of data (#10472) that was incomplete.  I'm going to verify that the data isn't useful and drop that from the analysis.

In [116]:
print(andr.loc[[10472]])
play = andr.drop(10472)

                                           App Category  Rating Reviews  \
10472  Life Made WI-Fi Touchscreen Photo Frame      1.9    19.0    3.0M   

         Size Installs Type     Price Content Rating             Genres  \
10472  1,000+     Free    0  Everyone            NaN  February 11, 2018   

      Last Updated Current Ver Android Ver  
10472       1.0.19  4.0 and up         NaN  


## Duplicate and English Apps 

As shown above, we have 10,841 apps listed on the google store and 7197 listed on the app store. Let's see how many of these apps are actually unique:

In [117]:
print("Unique Google Play Apps:")
print(play['App'].nunique())

Unique Google Play Apps:
9659


In [118]:
print("Unique App Store Apps:")
print(apple['track_name'].nunique())

Unique App Store Apps:
7195


Looks the like google play has 1181 duplicate app names, while the app store has two.  Let's create a dictionary to make sure that we have only one of each app in our data; we'll base which one to keep on which one has the most reviews. 

In [119]:
rev_max_play = {}

for i,row in play.iterrows():
    app = row['App']
    num_rev = float(row['Reviews'])
    
    if app in rev_max_play and rev_max_play[app] < num_rev:
        rev_max_play[app] = num_rev
        
    elif app not in rev_max_play:
        rev_max_play[app] = num_rev
        

rev_max_apple = {}

for i,row in apple.iterrows():
    appl = row['track_name']
    num_revi = float(row['rating_count_tot'])
    
    if app in rev_max_apple and rev_max_apple[appl] < num_revi:
        rev_max_apple[appl] = num_revi
        
    elif app not in rev_max_apple:
        rev_max_apple[appl] = num_revi
        
play_clean = []
already_added = []

for i,app in play.iterrows():
    name = app['App']
    n_reviews = float(app['Reviews'])
    
    if (rev_max_play[name] == n_reviews) and (name not in already_added):
        play_clean.append(app)
        already_added.append(name) 
        
apple_clean = []
already_added2 = []

for i,app in apple.iterrows():
    name = app['track_name']
    n_reviews = float(app['rating_count_tot'])
    
    if (rev_max_apple[name] == n_reviews) and (name not in already_added2):
        apple_clean.append(app)
        already_added2.append(name) 

if (len(play_clean)==(play['App'].nunique())) & (len(apple_clean)==(apple['track_name'].nunique())):
    print("All duplicates have been removed")

All duplicates have been removed


In [120]:
play_clean_df = pd.DataFrame(play_clean)
apple_clean_df = pd.DataFrame(apple_clean)

apple_clean_df.to_csv('/Users/atrain31/Desktop/Data/dq1/apple_clean.csv')
play_clean_df.to_csv('/Users/atrain31/Desktop/Data/dq1/play_clean.csv')

Now we have two lists, play_clean and apple_clean, which have only one of each app in the data.  The dictionaries worked as the cleaned lists are the exact same length as the number of unique values we originally found in the first step above. 

Further exploring the data, it would most likely be difficult to analyze data from non-english apps.  I'm going to weed those out by creating a function that identifies non-english language apps. 

In [121]:
def english_app(string):
    non_english_count = 0
    for character in string:
        number = ord(character)
        if number > 127:
            non_english_count += 1
        else:
            non_english_count += 0

    if non_english_count>2:
        return False
    else:
        return True
    
play_english = []
for app in play_clean:
    name = app[1]
    if english_app(name):
        play_english.append(app)
    else:
        None

apple_english = []
for app in apple_clean:
    name = app[1]
    if english_app(name):
        apple_english.append(app)
    else:
        None
        
print("# English Google Apps:")
print(len(play_english))

print("# English Apple Apps:")
print(len(apple_english))  

# English Google Apps:
9659
# English Apple Apps:
6153


We are now down to 9659 apps on google play and 6153 on the app store. 

Let's see how many of those apps were free - remember,  in the scenario I mentioned in the intro, I'm looking for free apps in an effort to develop an app based on ad revenue.  

## Pricing

Let's look at the pricing of the available apps.  I'm going to base our pricing on the "Price" column rather than the "Type" column and go from there (also the app store data does not have a Type column).  Let's see what we have for a range for both datasets:

In [122]:
play_english_df = pd.DataFrame(play_english)
apple_english_df = pd.DataFrame(apple_english)

print('Range of Pricing for Google Play Store:')
play_english_df['Price'].min(), play_english_df['Price'].max()

Range of Pricing for Google Play Store:


('$0.99', '0')

In [123]:
print('Range of Pricing for App Store:')
apple_english_df['price'].min(), apple_english_df['price'].max()

Range of Pricing for App Store:


(0.0, 299.99)

The Apple range looks good, but the Google range is off.  Let's fix that: 

In [124]:
play_english_df['Price'] = play_english_df['Price'].str.replace('$','')
play_english_df['Price'] = pd.to_numeric(play_english_df['Price'])

print('Corrected Range of Pricing for Google Play Store:')
play_english_df['Price'].min(), play_english_df['Price'].max()

Corrected Range of Pricing for Google Play Store:


(0.0, 400.0)

Before moving forward, I'd like to take a look at how prevalent paid apps are vs free apps and get an idea of how popular each type is with users. 

In [125]:
apple_english_df.to_csv('/Users/atrain31/Desktop/Data/dq1/apple_english.csv')
play_english_df.to_csv('/Users/atrain31/Desktop/Data/dq1/play_english_df.csv')

play_free = play_english_df[play_english_df['Price']==0.0]
apple_free = apple_english_df[apple_english_df['price']==0.0]

print("Number of Free Google Apps, App Store Apps:")
len(play_free), len(apple_free)

Number of Free Google Apps, App Store Apps


(8905, 3201)

There are 9659 total unique English apps in the Play store, 8905 of which are free; 6153 and 3201 in the app store.  I'd like to plug this into Tableau and see what it looks like:

In [31]:
%%HTML
<div class='tableauPlaceholder' id='viz1597409129902' style='position: relative'><noscript><a href='#'><img alt=' ' src='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;Go&#47;GooglePlayAppPriceRelativetoNumberofReviewsandAverageRating&#47;Sheet1&#47;1_rss.png' style='border: none' /></a></noscript><object class='tableauViz'  style='display:none;'><param name='host_url' value='https%3A%2F%2Fpublic.tableau.com%2F' /> <param name='embed_code_version' value='3' /> <param name='site_root' value='' /><param name='name' value='GooglePlayAppPriceRelativetoNumberofReviewsandAverageRating&#47;Sheet1' /><param name='tabs' value='no' /><param name='toolbar' value='yes' /><param name='static_image' value='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;Go&#47;GooglePlayAppPriceRelativetoNumberofReviewsandAverageRating&#47;Sheet1&#47;1.png' /> <param name='animate_transition' value='yes' /><param name='display_static_image' value='yes' /><param name='display_spinner' value='yes' /><param name='display_overlay' value='yes' /><param name='display_count' value='yes' /><param name='language' value='en' /><param name='filter' value='publish=yes' /></object></div>                <script type='text/javascript'>                    var divElement = document.getElementById('viz1597409129902');                    var vizElement = divElement.getElementsByTagName('object')[0];                    vizElement.style.width='100%';vizElement.style.height=(divElement.offsetWidth*0.75)+'px';                    var scriptElement = document.createElement('script');                    scriptElement.src = 'https://public.tableau.com/javascripts/api/viz_v1.js';                    vizElement.parentNode.insertBefore(scriptElement, vizElement);                </script>

The extreme popularity of ~50 apps totally skewers this graph, making it almost unusable, BUT - one thing we can see is that all of the very popular apps are free apps.  Let's further examine this: 

In [126]:
play_paid = play_english_df[play_english_df['Price']>0.0]

play_free['Reviews'] = play_free['Reviews'].astype(int)
play_paid['Reviews'] = play_paid['Reviews'].astype(int)

play_paid_popular = play_paid[play_paid['Reviews']>500000]
play_free_popular = play_free[play_free['Reviews']>500000]

print("Number of Paid Google Play Apps with >500,000 Reviews vs Free Play Apps with >500,000 Reviews: ")
len(play_paid_popular), len(play_free_popular)

Number of Paid Google Play Apps with >500,000 Reviews vs Free Play Apps with >500,000 Reviews: 


(1, 585)

In [127]:
play_paid_popular2 = play_paid[play_paid['Reviews']>100000]
play_free_popular2 = play_free[play_free['Reviews']>100000]

print("Number of Paid Google Play Apps with >100,000 Reviews vs Free Play Apps with >100,000 Reviews: ")
len(play_paid_popular2), len(play_free_popular2)

Number of Paid Google Play Apps with >100,000 Reviews vs Free Play Apps with >100,000 Reviews: 


(8, 1443)

In [132]:
print("Most Reviews for a free Google Play App:") 
print(play_free['Reviews'].max())
print("Most Reviews for a paid Google Play App:") 
print(play_paid['Reviews'].max())

Most Reviews for a free Google Play App:
78158306
Most Reviews for a paid Google Play App:
2376564


Wow!  This is a huge difference in popularity.  Not only do the most popular apps get tens of millions of reviews, but with the exception of only one, there are no paid apps with over 500,000 reviews.  And - only eight paid apps surpass 100,000 reviews, while 1443(!) surpass this amount.  

Let's look at the stats for the app store:

In [133]:
apple_paid = apple_english_df[apple_english_df['price']>0.0]

apple_free['rating_count_tot'] = apple_free['rating_count_tot'].astype(int)

apple_paid_popular = apple_paid[apple_paid['rating_count_tot']>100000]
apple_free_popular = apple_free[apple_free['rating_count_tot']>100000]

print("Number of Paid App Store Apps with >100,000 Reviews vs Free App Store Apps with >100,000 Reviews: ")
len(apple_paid_popular), len(apple_free_popular)

Number of Paid App Store Apps with >100,000 Reviews vs Free App Store Apps with >100,000 Reviews: 


(19, 199)

Again, there is a large difference, almost ten fold in the popularity of free apps vs. paid apps

Seeing as we're looking into an app with the highest earning potential, it would seem that our chances would be much better with a free app with revenue earned through advertisements.  

For the remainder of this analysis, I'll base our data on the free apps only. 

## Categories 

In [134]:
print("Free Google Play App Categories:")
play_free['Category'].unique()

Free Google Play App Categories:


array(['ART_AND_DESIGN', 'AUTO_AND_VEHICLES', 'BEAUTY',
       'BOOKS_AND_REFERENCE', 'BUSINESS', 'COMICS', 'COMMUNICATION',
       'DATING', 'EDUCATION', 'ENTERTAINMENT', 'EVENTS', 'FINANCE',
       'FOOD_AND_DRINK', 'HEALTH_AND_FITNESS', 'HOUSE_AND_HOME',
       'LIBRARIES_AND_DEMO', 'LIFESTYLE', 'GAME', 'FAMILY', 'MEDICAL',
       'SOCIAL', 'SHOPPING', 'PHOTOGRAPHY', 'SPORTS', 'TRAVEL_AND_LOCAL',
       'TOOLS', 'PERSONALIZATION', 'PRODUCTIVITY', 'PARENTING', 'WEATHER',
       'VIDEO_PLAYERS', 'NEWS_AND_MAGAZINES', 'MAPS_AND_NAVIGATION'],
      dtype=object)

In [135]:
print("Free App Store App Categories:")
apple_free['prime_genre'].unique()

Free App Store App Categories:


array(['Social Networking', 'Photo & Video', 'Games', 'Music',
       'Reference', 'Health & Fitness', 'Weather', 'Utilities', 'Travel',
       'Shopping', 'News', 'Navigation', 'Lifestyle', 'Entertainment',
       'Food & Drink', 'Sports', 'Book', 'Finance', 'Education',
       'Productivity', 'Business', 'Catalogs', 'Medical'], dtype=object)

The differences in categories between the two operating systems prevents us really from combining the data together.  But - we can still use the categories in our analysis separately.  

## Number of Reviews(Ratings)/Installs 

The top Google Play apps by # of installs:

In [140]:
#Cleaning the Installs column:
play_free['Installs'] = play_free['Installs'].str.replace('+','')
play_free['Installs'] = play_free['Installs'].str.replace(',','')
play_free['Installs'] = play_free['Installs'].astype(int)

play_free.sort_values('Installs',ascending=False).head(10)

App       Category  Rating   Reviews  \
336              WhatsApp Messenger  COMMUNICATION     4.4  69119316   
391   Skype - free IM & video calls  COMMUNICATION     4.1  10484169   
2604                      Instagram         SOCIAL     4.5  66577446   
451                           Gmail  COMMUNICATION     4.3   4604483   
464                        Hangouts  COMMUNICATION     4.0   3419513   
3665                        YouTube  VIDEO_PLAYERS     4.3  25655305   
2554                        Google+         SOCIAL     4.2   4831125   
1872                 Subway Surfers           GAME     4.5  27725352   
2884                  Google Photos    PHOTOGRAPHY     4.5  10859051   
3523                   Google Drive   PRODUCTIVITY     4.4   2731211   

                    Size    Installs  Type  Price Content Rating  \
336   Varies with device  1000000000  Free    0.0       Everyone   
391   Varies with device  1000000000  Free    0.0       Everyone   
2604  Varies with device  1000000000  Free    0.0           Teen   
451   Varies with device  1000000000  Free    0.0       Everyone   
464   Varies with device  1000000000  Free    0.0       Everyone   
3665  Varies with device  1000000000  Free    0.0           Teen   
2554  Varies with device  1000000000  Free    0.0           Teen   
1872                 76M  1000000000  Free    0.0   Everyone 10+   
2884  Varies with device  1000000000  Free    0.0       Everyone   
3523  Varies with device  1000000000  Free    0.0       Everyone   

                       Genres    Last Updated         Current Ver  \
336             Communication  August 3, 2018  Varies with device   
391             Communication  August 3, 2018  Varies with device   
2604                   Social   July 31, 2018  Varies with device   
451             Communication  August 2, 2018  Varies with device   
464             Communication   July 21, 2018  Varies with device   
3665  Video Players & Editors  August 2, 2018  Varies with device   
2554                   Social   July 26, 2018  Varies with device   
1872                   Arcade   July 12, 2018              1.90.0   
2884              Photography  August 6, 2018  Varies with device   
3523             Productivity  August 6, 2018  Varies with device   

             Android Ver  
336   Varies with device  
391   Varies with device  
2604  Varies with device  
451   Varies with device  
464   Varies with device  
3665  Varies with device  
2554  Varies with device  
1872          4.1 and up  
2884  Varies with device  
3523  Varies with device

Here we can see that there are several dominant apps in various categories that are outliers and don't represent 'regular' apps. This is better illustrated below:

In [41]:
%%HTML
<div class='tableauPlaceholder' id='viz1597608679877' style='position: relative'><noscript><a href='#'><img alt=' ' src='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;Go&#47;GooglePlayOutliers&#47;Sheet4&#47;1_rss.png' style='border: none' /></a></noscript><object class='tableauViz'  style='display:none;'><param name='host_url' value='https%3A%2F%2Fpublic.tableau.com%2F' /> <param name='embed_code_version' value='3' /> <param name='site_root' value='' /><param name='name' value='GooglePlayOutliers&#47;Sheet4' /><param name='tabs' value='no' /><param name='toolbar' value='yes' /><param name='static_image' value='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;Go&#47;GooglePlayOutliers&#47;Sheet4&#47;1.png' /> <param name='animate_transition' value='yes' /><param name='display_static_image' value='yes' /><param name='display_spinner' value='yes' /><param name='display_overlay' value='yes' /><param name='display_count' value='yes' /><param name='language' value='en' /><param name='filter' value='publish=yes' /></object></div>                <script type='text/javascript'>                    var divElement = document.getElementById('viz1597608679877');                    var vizElement = divElement.getElementsByTagName('object')[0];                    vizElement.style.width='100%';vizElement.style.height=(divElement.offsetWidth*0.75)+'px';                    var scriptElement = document.createElement('script');                    scriptElement.src = 'https://public.tableau.com/javascripts/api/viz_v1.js';                    vizElement.parentNode.insertBefore(scriptElement, vizElement);                </script>

Let's see what happens when we sort by the mean # of installs for each category:

In [155]:
play_free.groupby('Category')['Installs'].mean().sort_values(ascending=False).reset_index().head(10)

Category      Installs
0       COMMUNICATION  3.832263e+07
1       VIDEO_PLAYERS  2.457395e+07
2              SOCIAL  2.325365e+07
3         PHOTOGRAPHY  1.777202e+07
4        PRODUCTIVITY  1.673896e+07
5                GAME  1.555200e+07
6    TRAVEL_AND_LOCAL  1.398408e+07
7       ENTERTAINMENT  1.164071e+07
8               TOOLS  1.078701e+07
9  NEWS_AND_MAGAZINES  9.401636e+06

Now let's see what happens when we take out the top 5% of installed apps and the bottom 5% and we revisit the mean, we can see that the the order has been displaced once the extremely popular apps are removed.  I'd probably stay away from the video players, social, and communication categories as making an app in those categories would face stiff competition.  

In [156]:
iqr = play_free[play_free['Installs'].between(play_free['Installs'].quantile(.05), play_free['Installs'].quantile(.95), inclusive=True)]
iqr.groupby('Category')['Installs'].mean().sort_values(ascending=False).reset_index().head(10)

Category      Installs
0     ENTERTAINMENT  3.808684e+06
1       PHOTOGRAPHY  3.453283e+06
2              GAME  3.306546e+06
3          SHOPPING  3.074517e+06
4     COMMUNICATION  2.476439e+06
5           WEATHER  2.465977e+06
6     VIDEO_PLAYERS  2.387278e+06
7            SOCIAL  2.146381e+06
8            SPORTS  2.081032e+06
9  TRAVEL_AND_LOCAL  2.055750e+06

In [141]:
print("Number of Apps When Taking Out the Highest and Lowest 5% of Number of Reviewed Apps vs All Free Apps:")
len(iqr), len(play_free)

Number of Apps When Taking Out the Highest and Lowest 5% of Number of Reviewed Apps vs All Free Apps:


(8032, 8905)

Top 10 Apps After Adjusting for the Outliers:

In [142]:
iqr.sort_values('Installs',ascending=False).head(10)

App          Category  \
5948                             PBA® Bowling Challenge            SPORTS   
3149                    Despegar.com Hotels and Flights  TRAVEL_AND_LOCAL   
3268                          Google app for Android TV             TOOLS   
3267                                   AT&T Smart Wi-Fi             TOOLS   
3264                                    HTC Lock Screen             TOOLS   
3263                                              Phone             TOOLS   
3262                                              Clear             TOOLS   
3260  File Manager -- Take Command of Your Files Easily             TOOLS   
3257    Files Go by Google: Free up space on your phone             TOOLS   
3256                      ZenUI Keyboard – Emoji, Theme             TOOLS   

      Rating  Reviews                Size  Installs  Type  Price  \
5948     4.3   242096                 87M  10000000  Free    0.0   
3149     4.4   150932                 27M  10000000  Free    0.0   
3268     3.0       66  Varies with device  10000000  Free    0.0   
3267     3.9    18513                6.1M  10000000  Free    0.0   
3264     4.1    28250  Varies with device  10000000  Free    0.0   
3263     3.6    45483  Varies with device  10000000  Free    0.0   
3262     3.1    24151                 15M  10000000  Free    0.0   
3260     4.2   127223                7.6M  10000000  Free    0.0   
3257     4.6   315585                8.5M  10000000  Free    0.0   
3256     4.5   537554  Varies with device  10000000  Free    0.0   

     Content Rating          Genres    Last Updated              Current Ver  \
5948       Everyone          Sports  August 2, 2018                    3.4.4   
3149       Everyone  Travel & Local   July 30, 2018                    9.1.0   
3268       Everyone           Tools   July 19, 2018       Varies with device   
3267       Everyone           Tools  April 26, 2018                      3.2   
3264       Everyone           Tools  March 21, 2017       Varies with device   
3263       Everyone           Tools   July 30, 2018       Varies with device   
3262       Everyone           Tools  August 6, 2018                   11.7.0   
3260       Everyone           Tools  August 3, 2018  v7.0.7.1.0625.1_06_0629   
3257       Everyone           Tools   July 18, 2018            1.0.204375696   
3256       Everyone           Tools  August 1, 2018       Varies with device   

             Android Ver  
5948          4.1 and up  
3149          4.1 and up  
3268  Varies with device  
3267          4.1 and up  
3264  Varies with device  
3263  Varies with device  
3262          4.1 and up  
3260          5.0 and up  
3257          5.0 and up  
3256  Varies with device

I think that the number of times an app was reviewed is a nice gauge to see how popular an app was (it's also necessary when looking at the Apple data as there is no data on the amount of installs). On Google it is the 'Reviews' column and on Apple it is the 'rating_count_tot' column.  I want to make sure the results aren't skewed with apps that were rated a limited number of times, so I'm going to omit the lowest quartile when looking at the number of reviews/ratings.  

In [143]:
play_free2 = iqr
play_free2['Reviews'] = play_free2['Reviews'].astype(int)

print("Google Quartiles")
play_free2.Reviews.quantile([0.25,0.5,0.75])

Quartiles


0.25       42.00
0.50     1375.50
0.75    27506.75
Name: Reviews, dtype: float64

In [144]:
play_english = play_free2[play_free2['Reviews']>42]

In [146]:
print("Apple Quantile")
apple_free.rating_count_tot.quantile([0.25,0.5,0.75,0.95])

Apple Quantile


0.25       147.0
0.50      1204.0
0.75      9975.0
0.95    119316.0
Name: rating_count_tot, dtype: float64

In [52]:
apple_english = apple_free[(apple_free['rating_count_tot']>147) & (apple_free['rating_count_tot']<119316)]

Again, let's look at the before and after when the top and bottom 5% are removed, this time with the Apple Store apps:

In [157]:
print("Before:")
apple_free.groupby('prime_genre')['rating_count_tot'].mean().sort_values(ascending=False).reset_index().head(10)

Before:


prime_genre  rating_count_tot
0         Navigation      86090.333333
1          Reference      79350.470588
2  Social Networking      71548.349057
3              Music      57326.530303
4            Weather      52279.892857
5               Book      46384.916667
6       Food & Drink      33333.923077
7            Finance      32367.028571
8      Photo & Video      28441.543750
9             Travel      28243.800000

In [158]:
print("After:")
apple_english = apple_free[(apple_free['rating_count_tot']>147) & (apple_free['rating_count_tot']<119316)]
apple_english.groupby('prime_genre')['rating_count_tot'].mean().sort_values(ascending=False).reset_index().head(10)

After:


prime_genre  rating_count_tot
0               Book      38066.750000
1            Weather      27613.500000
2            Finance      19415.363636
3           Shopping      18160.338462
4       Food & Drink      17887.647059
5  Social Networking      16858.388235
6       Productivity      16823.666667
7             Sports      15323.162791
8               News      15225.785714
9          Utilities      14104.490909

In [55]:
#DL these clean files to desktop for Tableau use 

apple_english.to_csv('/Users/atrain31/Desktop/Data/dq1/apple_english.csv')
play_english.to_csv('/Users/atrain31/Desktop/Data/dq1/play_english.csv')

## Ratings

In [56]:
%%HTML

<div class='tableauPlaceholder' id='viz1597420646361' style='position: relative'><noscript><a href='#'><img alt=' ' src='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;Go&#47;GooglePlayAppRatingvs_Installs&#47;Sheet2&#47;1_rss.png' style='border: none' /></a></noscript><object class='tableauViz'  style='display:none;'><param name='host_url' value='https%3A%2F%2Fpublic.tableau.com%2F' /> <param name='embed_code_version' value='3' /> <param name='site_root' value='' /><param name='name' value='GooglePlayAppRatingvs_Installs&#47;Sheet2' /><param name='tabs' value='no' /><param name='toolbar' value='yes' /><param name='static_image' value='https:&#47;&#47;public.tableau.com&#47;static&#47;images&#47;Go&#47;GooglePlayAppRatingvs_Installs&#47;Sheet2&#47;1.png' /> <param name='animate_transition' value='yes' /><param name='display_static_image' value='yes' /><param name='display_spinner' value='yes' /><param name='display_overlay' value='yes' /><param name='display_count' value='yes' /><param name='language' value='en' /><param name='filter' value='publish=yes' /></object></div>                <script type='text/javascript'>                    var divElement = document.getElementById('viz1597420646361');                    var vizElement = divElement.getElementsByTagName('object')[0];                    vizElement.style.width='100%';vizElement.style.height=(divElement.offsetWidth*0.75)+'px';                    var scriptElement = document.createElement('script');                    scriptElement.src = 'https://public.tableau.com/javascripts/api/viz_v1.js';                    vizElement.parentNode.insertBefore(scriptElement, vizElement);                </script>

Conventional wisdom would point to app ratings increasing as the number of installs increases, and this chart reflects that thought.  Obviously a poorly reviewed product will decrease it's popularity.  Going forward we can assume that a well-made and well-reviewed app will translate into higher downloads. 

# Further Analysis

Now that we have the list of apps that are both in english and free, we can really dive into some potential trends.  I'd like to check on how many reviews per category are present, so I'll create two dictionaries for each dataset: 1) the count for the number of apps for each category and 2) the number of reviews for each category.  

In [57]:
# Google Play
categories_two  = list(set(play_english['Category']))
categ_inst_dict = {}
categ_count = {}
categ_avg_rating = {}

for cat in categories_two:

    new_df = play_english[play_english['Category']==cat]
    
    count = 0
    count += len(new_df)
    sum_of_installs = new_df['Reviews'].sum()
    categ_inst_dict[cat] = sum_of_installs
    categ_count[cat] = count
    
    ratings_count = 0
    ratings_df = new_df.dropna(subset=['Rating'])
    ratings_count = len(ratings_df)
    categ_avg_rating[cat] = (ratings_df['Rating'].sum())/ratings_count

    
#App Store
categories_three  = list(set(apple_english['prime_genre']))
a_categ_inst_dict = {}
a_categ_count = {}

for cat in categories_three:
    a_count = 0
    a_new_df = apple_english[apple_english['prime_genre']==cat]
    a_new_df['rating_count_tot'] = a_new_df['rating_count_tot'].astype(int)
    a_count += len(a_new_df)
    a_sum_of_installs = a_new_df['rating_count_tot'].sum()
    a_categ_inst_dict[cat] = a_sum_of_installs
    a_categ_count[cat] = a_count

The Google data, sorted by % of Total Apps: 

In [151]:
new_play_df = pd.DataFrame(categories_two,columns=['Category'])
new_play_df['Count'] = new_play_df['Category'].map(categ_count)
new_play_df['# of Reviews'] = new_play_df['Category'].map(categ_inst_dict)
new_play_df['Avg # Reviews Per App'] = (new_play_df['# of Reviews'] / new_play_df['Count'])
new_play_df['% of Total Apps'] = (new_play_df['Count'] / new_play_df['Count'].sum())
new_play_df.sort_values('% of Total Apps',ascending=False).head()

Category  Count  # of Reviews  Avg # Reviews Per App  % of Total Apps
21     FAMILY   1172      81417526           69468.878840         0.194781
7        GAME    669     107215876          160262.893871         0.111185
13      TOOLS    488      26208389           53705.715164         0.081104
8     FINANCE    243      11631014           47864.255144         0.040386
6   LIFESTYLE    214       8912231           41645.939252         0.035566

The Google data, sorted by Avg # Reviews Per App: 

In [150]:
new_play_df.sort_values('Avg # Reviews Per App',ascending=False).head()

Category  Count  # of Reviews  Avg # Reviews Per App  % of Total Apps
7            GAME    669     107215876          160262.893871         0.111185
18         SOCIAL    154      17539773          113894.629870         0.025594
28    PHOTOGRAPHY    180      18485694          102698.300000         0.029915
5   ENTERTAINMENT     76       7482110           98448.815789         0.012631
1   COMMUNICATION    166      15789565           95117.861446         0.027588

The Apple Data:

In [152]:
new_apple = pd.DataFrame(categories_three,columns=['Category'])
new_apple['Count'] = new_apple['Category'].map(a_categ_count)
new_apple['# of Ratings'] = new_apple['Category'].map(a_categ_inst_dict)
new_apple['Avg # Ratings Per App'] = (new_apple['# of Ratings'] / new_apple['Count'])
new_apple['% of Total Apps'] = (new_apple['Count'] / new_apple['Count'].sum())
new_apple.sort_values('% of Total Apps',ascending=False).head()

Category  Count  # of Ratings  Avg # Ratings Per App  \
7               Games   1274      14150897           11107.454474   
17      Entertainment    191       2357265           12341.701571   
0       Photo & Video    121       1539030           12719.256198   
15  Social Networking     85       1432963           16858.388235   
4           Education     75        538759            7183.453333   

    % of Total Apps  
7          0.569258  
17         0.085344  
0          0.054066  
15         0.037980  
4          0.033512

In [153]:
new_apple.sort_values('Avg # Ratings Per App',ascending=False).head()

Category  Count  # of Ratings  Avg # Ratings Per App  % of Total Apps
11          Book      8        304534           38066.750000         0.003575
3        Weather     10        276135           27613.500000         0.004468
2        Finance     22        427138           19415.363636         0.009830
5       Shopping     65       1180422           18160.338462         0.029044
16  Food & Drink     17        304090           17887.647059         0.007596

## Conclusion

Questions and ideas to consider when developing a new app:

- Filtering out the extremely popular apps gives a clearer picture of how to look at which categories perform the best on each platform.  Minus a revolutionary idea, it probably wouldn't make sense to go up against an app in one of these categories.  

- Cross-checking the categories which are the most popular with the ones that have the most reviews per app (Games on Google and Social on Apple) would show that these categories would be the safer bet.  

- The app development team would have to make a decision, based on app idea, as to finding an area that is not too saturated (such as Family on Google) vs an area that is popular but still has some potential (such as Books on ios).  
- Does the app idea cost continual money to maintain? ie - a paid API to a sports betting site

- Would it be possible to make an app focused on a potentially popular category based on an exisiting popular idea?  

With all of these taken into consideration the data suggests that something in the Books category may prove lucrative - it is a well reviewed category but still has room for growth.  If users were able to interact in a social setting and discuss books, review/recommend books, and maybe even participate in games related to books, I think that something like this would check a lot of the boxes for the popular categories of apps without infringing on the existing apps.  Most of the data would be user-generated and new content is consistently being generated.  